method.rb

```diff
@@ -0,0 +1,169 @@
+module Enumerable
+  def my_each
+    i = 0
+    arr = to_a
+    return to_enum(:my_each) unless block_given?
+
+    while i < arr.size
+      yield arr[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    i = 0
+    arr = to_a
+    return to_enum(:my_each_with_index) unless block_given?
+
+    while i < arr.size
+      yield arr[i], i
+      i += 1
+    end
+
+    self
+  end
+
+  def my_select
+    new_arr = []
+    return to_enum(:my_select) unless block_given?
+
+    my_each do |num|
+      new_arr.push(num) if yield num
+    end
+
+    new_arr
+  end
+
+  def my_all?(arg = nil)
+    stat = true
+
+    if arg
+      if arg.class == Class
+        my_each do |num|
+          stat = num.is_a?(arg)
+        end
+      else
+        my_each do |num|
+          stat = num == arg
+        end
+      end
+
+    elsif block_given?
+      my_each do |num|
+        stat = false unless yield num
+      end
+    else
+      return stat
+    end
+    stat
+  end
+
+  def my_any?(arg = nil)
+    stat = false
+
+    if arg
+      if arg.class == Class
+        my_each do |num|
+          stat = num.is_a?(arg)
+        end
+      else
+        my_each do |num|
+          stat = num == arg
+        end
+      end
+
+    elsif block_given?
+      my_each do |num|
+        stat = true if yield num
+      end
+    else
+      my_each do |num|
+        bool = !num
+        stat = !bool
+      end
+    end
+    stat
+  end
```

Comment:
`my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- when a Regex is passed as an argument returns false if none of the collection matches the Regex
```ruby
%w[dog door rod blade].my_any?(/t/)    #should return false.
%w[dog door rod blade].my_any?(/d/)   #should return true.
```
- when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
```ruby
[1, 2, 3].my_any?(3)   #should return true
[1, 2, 2].my_any?(3)   #should return false
```